### PR TITLE
Auto-enable more integrations

### PR DESCRIPTION
**Pull request by sentrivana**
Auto-enable more integrations.

Closes https://github.com/getsentry/sentry-python/issues/2630

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - The `reraise` function was moved from `sentry_sdk._compat` to `sentry_sdk.utils`.
 - Moved the contents of `tracing_utils_py3.py` to `tracing_utils.py`. The `start_child_span_decorator` is now in `sentry_sdk.tracing_utils`.
 - The actual implementation of `get_current_span` was moved to `sentry_sdk.tracing_utils`. `sentry_sdk.get_current_span` is still accessible as part of the top-level API.
+- Additional integrations will now be activated automatically if the SDK detects the respective package is installed: Ariadne, ARQ, asyncpg, Chalice, clickhouse-driver, GQL, Graphene, huey, Loguru, pymongo, Quart, Starlite, Strawberry.
```
Comment (szokeasaurusrex) [inline MIGRATION_GUIDE.md:13]: What is the reasoning for the inconsistent capitalization of the integration names? Does the capitalization here match the docs?

File: sentry_sdk/integrations/starlite.py [hunk 2]
```diff
@@ -15,6 +14,7 @@
     from starlite.plugins.base import get_plugin_for_value  # type: ignore
     from starlite.routes.http import HTTPRoute  # type: ignore
     from starlite.utils import ConnectionDataExtractor, is_async_callable, Ref  # type: ignore
+    from pydantic import BaseModel  # type: ignore
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/starlite.py:17]: Does `starlite` depend on `pydantic`? Trying to understand why we need to move this line into the `try` block

File: sentry_sdk/integrations/strawberry.py [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
 from inspect import isawaitable
+
+try:
+    from functools import cached_property
+except ImportError:
+    # The strawberry integration requires Python 3.8+. functools.cached_property
+    # was added in 3.8, so this check is technically not needed, but since this
+    # is an auto-enabling integration, we might get to executing this import in
+    # lower Python versions, so we need to deal with it.
+    cached_property = None  # type: ignore
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/strawberry.py:11]: Why not raise `DidNotEnable` exception here, with a warning that the Python version is too low?

**Review by szokeasaurusrex — COMMENTED** [review]
Have some questions

File: sentry_sdk/integrations/starlite.py (hunk 2 above)
Comment (sentrivana) [reply sentry_sdk/integrations/starlite.py:17]: Yes, it uses `pydantic` for valdidation and parsing.

File: MIGRATION_GUIDE.md (hunk 1 above)
Comment (sentrivana) [reply MIGRATION_GUIDE.md:13]: Tried to be in line with how the specific package usually styles the name in their docs/on their GH/etc.

`pymongo` is wrong though -- will update that

File: sentry_sdk/integrations/strawberry.py (hunk 3 above)
Comment (sentrivana) [reply sentry_sdk/integrations/strawberry.py]: That's a good idea. Had to move the `functools` import until after the `sentry_sdk` import that imports `DidNotEnable`, but it's still a nicer solution.

**Review by szokeasaurusrex — APPROVED** [review]
Cool, thanks for addressing my feedback!

**Pull request merged**